src/ui/public/notify/toasts/global_toast_list.js

```diff
@@ -0,0 +1,133 @@
+import React, {
+  Component,
+} from 'react';
+import PropTypes from 'prop-types';
+import 'ngreact';
+import { uiModules } from 'ui/modules';
+
+import {
+  EuiGlobalToastList,
+  EuiGlobalToastListItem,
+  EuiToast,
+} from '@elastic/eui';
+
+export const TOAST_FADE_OUT_MS = 250;
+
+export class GlobalToastList extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      toastIdToDismissedMap: {}
+    };
+
+    this.timeoutIds = [];
+    this.toastsScheduledForDismissal = {};
+  }
+
+  static propTypes = {
+    toasts: PropTypes.array,
+    dismissToast: PropTypes.func.isRequired,
+    toastLifeTimeMs: PropTypes.number.isRequired,
+  };
+
+  static defaultProps = {
+    toasts: [],
+  };
+
+  scheduleAllToastsForDismissal = () => {
+    this.props.toasts.forEach(toast => {
+      if (!this.toastsScheduledForDismissal[toast.id]) {
+        this.scheduleToastForDismissal(toast);
+      }
+    });
+  };
+
+  scheduleToastForDismissal = (toast, isImmediate = false) => {
+    const toastLifeTimeMs = isImmediate ? 0 : this.props.toastLifeTimeMs;
+
+    // Start fading the toast out once its lifetime elapses.
+    this.timeoutIds.push(setTimeout(() => {
+      this.startDismissingToast(toast);
+    }, toastLifeTimeMs));
+
+    // Remove the toast after it's done fading out.
+    this.timeoutIds.push(setTimeout(() => {
+      this.props.dismissToast(toast);
+      this.setState(prevState => {
+        const toastIdToDismissedMap = { ...prevState.toastIdToDismissedMap };
+        delete toastIdToDismissedMap[toast.id];
+
+        return {
+          toastIdToDismissedMap,
+        };
+      });
+    }, toastLifeTimeMs + TOAST_FADE_OUT_MS));
+  };
+
+  startDismissingToast(toast) {
+    this.setState(prevState => {
+      const toastIdToDismissedMap = { ...prevState.toastIdToDismissedMap };
```

Comment:
FYI, this can be written as:

```
const toastIdToDismissedMap = {
   ...prevState.toastIdToDismissedMap,
   [toast.id]: true,
};
```

No need to change, but I sort of like it better this way and wanted to suggest in case you weren't aware.